src/plugins/PluginBase.h

```diff
@@ -140,6 +144,45 @@ class PluginBase {
 
     /** @brief return the name of the plugin */
     virtual std::string getPluginName()  = 0;
+
+	/**
+	 * Register callbacks to callbackMap.
+	 * This allows to make functions of a plugin accessible to other plugins.
+	 * New callbacks should be added to callbackMap.
+	 * Example syntax:
+	 * - register a function that returns a local value:
+	 * \code
+	 *   callbackMap["getMyLocalValue"] = [this] { return _myLocalValue; };
+	 * \endcode
+	 * - register a function that calls a local function and returns its return value:
+	 * \code
+	 *   callbackMap["callMyFunct"] = [this] { return myFunct(); };
+	 * \endcode
+	 * @param callbackMap Add callbacks to this map.
+	 */
+	virtual void registerCallbacks(std::map<std::string, FunctionWrapper>& callbackMap) {
+		// Empty by default.
+	}
+
+	/**
+	 * Save callbacks from the callbackMap locally.
+	 * This allows a plugin to call functions from other plugins.
+	 * Example syntax:
+	 * - store a function that returns an unsigned long:
+	 * \code
+	 *   std::function<unsigned long(void)> myFunction;
+	 *   myFunction = callbackMap.at("getSomeLocalValue").get<unsigned long>();
+	 * \endcode
+	 * - store a function that calls some function of another plugin with an input value (int):
+	 * \code
+	 *   std::function<void(int)> myFunction;
+	 *   myFunction = callbackMap.at("doSth").get<void, int>();
+	 * \endcode
+	 * @param callbackMap Get callbacks from this map.
+	 */
+	virtual void saveCallbacksLocally(const std::map<std::string, FunctionWrapper>& callbackMap) {

```

Comment:
This function name is highly misleading as it actually does nothing. It hints at what it could be used for (to save callbacks locally) but it could also be used in other ways. Nevertheless, this is what the documentation is for. Rename maybe to `accessAllCallbacks` or something like that.